ozaria/site/components/teacher-dashboard/modals/ModalAssignContent/ModalAssignForm.vue

```diff
@@ -34,15 +35,52 @@ export default {
       noty({ text: $.i18n.t('teacher_dashboard.select_student_first'), layout: 'center', type: 'information', killer: true, timeout: 8000 })
       this.$emit('close')
     }
+    this.generateCourses()
```

Comment:
I would instead do something like
this.groupedCourses = generateGroupedCourses() so that function could be re-used and tested independently of this vue class.